sensor/common/sensor/sensor.go

```diff
@@ -436,9 +436,13 @@ func (s *Sensor) communicationWithCentralWithRetries(centralReachable *concurren
 			// Connection is up, we can try to create a new central communication
 			s.changeState(common.SensorComponentEventCentralReachable)
 		case <-s.centralConnectionFactory.StopSignal().WaitC():
+			// Save the error before retrying
+			err := wrapOrNewError(s.centralConnectionFactory.StopSignal().Err(), "communication stopped")
+			// Reset the ok and stop signals
+			s.centralConnectionFactory.Reset()
```

Comment:
I think this is now redundant and might lead to confusion for the readers. I'd also suggest to remove `s.centralConnectionFactory.Reset()` down below around L476.

Reply:
Makes sense